Realm/ObjectServerTests/RLMSyncTestCase.mm

```diff
@@ -294,20 +294,27 @@ - (RLMSyncUser *)makeAdminUser:(NSString *)userName password:(NSString *)passwor
 }
 
 - (void)waitForDownloadsForUser:(RLMSyncUser *)user url:(NSURL *)url {
-    [self waitForDownloadsForUser:user url:url error:nil];
+    [self waitForDownloadsForUser:user url:url expectation:nil error:nil];
 }
 
-- (void)waitForDownloadsForUser:(RLMSyncUser *)user url:(NSURL *)url error:(NSError **)error {
+- (void)waitForDownloadsForUser:(RLMSyncUser *)user
+                            url:(NSURL *)url
+                    expectation:(XCTestExpectation *)expectation
+                          error:(NSError **)error {
     RLMSyncSession *session = [user sessionForURL:url];
     NSAssert(session, @"Cannot call with invalid URL");
-    XCTestExpectation *ex = [self expectationWithDescription:@"Download waiter expectation"];
+    XCTestExpectation *ex = expectation ?: [self expectationWithDescription:@"Download waiter expectation"];
     __block NSError *theError = nil;
-    [session waitForDownloadCompletionOnQueue:dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_LOW, 0)
-                                     callback:^(NSError *err){
-                                         theError = err;
-                                         [ex fulfill];
-                                     }];
-    [self waitForExpectationsWithTimeout:10.0 handler:nil];
+    BOOL queued = [session waitForDownloadCompletionOnQueue:dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_LOW, 0)
+                                                   callback:^(NSError *err){
+                                                       theError = err;
+                                                       [ex fulfill];
+                                                   }];
+    if (!queued) {
+        XCTFail(@"Download waiter did not queue; session was invalid or errored out.");
+        return;
+    }
+    [self waitForExpectations:@[ex] timeout:20.0];
```

Comment:
This one (and the one below) do seem to be making a difference. Unfortunately, everything here is guesswork since the tests never fail in this way on my MBP and they don't even fail all the time on our CI machines. I earlier made changes to try to narrow down the cause of this expectation failing and reached the conclusion that the issue simply was `waitForDownloads` not being given enough time to complete.